doc/design/milestone-0.0.7/csv-generation.md

```diff
@@ -0,0 +1,198 @@
+  ...
+  appsv1 "k8s.io/api/apps/v1"
+  rbacv1beta1 "k8s.io/api/rbac/v1beta1"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+  "github.com/operator-framework/operator-lifecycle-manager/pkg/controller/install"
+)
+
+// CSVInstallStrategyUpdate embeds the OLM's install strategy spec.
+type CSVInstallStrategyUpdate struct {
+	*install.StrategyDetailsDeployment
+  
+  // Future utility fields go here.
+}
+
+// getLocalInstallStrategyCache retrieves the local cache singleton and returns the install strategy cache.
+func getLocalInstallStrategyCache() *CSVInstallStrategyUpdate {
+	factory := getLocalCacheFactory()
+	return factory.InstallStrategyCache
+}
+
+// AddDeploymentSpecToCSVInstallStrategyUpdate adds an RBAC Role to the local cache singletons' permissions.
+func AddRoleToCSVInstallStrategyUpdate(yamlDoc []byte) error {
+	localInstallStrategyUpdate := getLocalInstallStrategyCache()
+
+	newRBACRole := new(rbacv1beta1.Role)
+	_ = yaml.Unmarshal(yamlDoc, newRBACRole)
+
+	newPermissions := install.StrategyDeploymentPermissions{
+		ServiceAccountName: newRole.ObjectMeta.Name,
+		Rules:              newRole.Rules,
+	}
+	localInstallStrategyUpdate.Permissions = append(localInstallStrategyUpdate.Permissions, newPermissions)
+
+	return nil
+}
+
+// AddDeploymentSpecToCSVInstallStrategyUpdate adds a Deployment to the local cache singletons' install strategy.
+func AddDeploymentSpecToCSVInstallStrategyUpdate(yamlDoc []byte) error {
+	localInstallStrategyUpdate := getLocalInstallStrategyCache()
+
+	newDeployment := new(appsv1.Deployment)
+	_ = yaml.Unmarshal(yamlDoc, newDeployment)
+  
+	newDeploymentSpec := install.StrategyDeploymentSpec{
+		Name: newDeployment.ObjectMeta.Name,
+		Spec: newDeployment.Spec,
+	}
+	localInstallStrategyUpdate.DeploymentSpecs = append(localInstallStrategyUpdate.DeploymentSpecs, newDeploymentSpec)
+
+	return nil
+}
+
+// Apply applies cached updates in CSVInstallStrategyUpdate to the appropriate csv fields.
+func (us *CSVInstallStrategyUpdate) Apply(csv *v1alpha1.ClusterServiceVersion) error {
+	// Get install strategy from csv.
+	var resolver *install.StrategyResolver
+	strat, _ := resolver.UnmarshalStrategy(csv.Spec.InstallStrategy)
+	installStrat, _ := strat.(*install.StrategyDetailsDeployment)
+
+	// Update permissions and deployments with custom field update methods.
+	us.updatePermissions(installStrat)
+	us.updateDeploymentSpecs(installStrat)
+
+	// Re-serialize permissions into csv install strategy.
+	updatedStrat, _ := json.Marshal(installStrat)
+	csv.Spec.InstallStrategy.StrategySpecRaw = updatedStrat
+
+	return nil
+}
+```
+
+### User-defined yaml objects
+
+Many CSV component objects cannot be populated using generated, non-SDK-specific manifests. These objects are mostly human-written, English metadata about the operator and various CRD's. A file, `describe.yaml`, with the following format should be written by a user with which the SDK can populate CSV objects:
```

Comment:
Where would this file live? Would it be a known location or would it be passed as a flag to `operator-sdk generate olm-catalog`?

Reply:
I'll add a sentence on location. I expect it to live either in the top level directory or in `deploy`. Any opinion on either?

Reply:
Either sound good, guess `deploy` makes a bit more sense.